src/lib/Microsoft.Health.Extensions.Fhir.R4/BearerTokenAuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,54 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using EnsureThat;
+using Microsoft.Health.Common.Telemetry;
+using Microsoft.Health.Extensions.Fhir.Telemetry.Metrics;
+using Microsoft.Health.Logging.Telemetry;
+
+namespace Microsoft.Health.Extensions.Fhir
+{
+    public class BearerTokenAuthorizationMessageHandler : DelegatingHandler
+    {
+        public BearerTokenAuthorizationMessageHandler(Uri uri, TokenCredential tokenCredentialProvider, ITelemetryLogger logger)
+        {
+            TokenCredential = EnsureArg.IsNotNull(tokenCredentialProvider, nameof(tokenCredentialProvider));
+            Uri = EnsureArg.IsNotNull(uri);
+            Scopes = new string[] { Uri + ".default" };
+            Logger = logger;
+        }
+
+        private ITelemetryLogger Logger { get; }
+
+        private TokenCredential TokenCredential { get; }
+
+        private Uri Uri { get; }
+
+        private string[] Scopes { get; }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            var requestContext = new TokenRequestContext(Scopes);
+            var accessToken = await TokenCredential.GetTokenAsync(requestContext, CancellationToken.None);
```

Comment:
Thanks for calling this out. We discussed this offline and I've created a [WI](https://microsofthealth.visualstudio.com/Health/_workitems/edit/88540) that can be taken up in a subsequent PR to cache the results of AquireServiceTokenAsync(). This will enable us to leverage the token for its full lifetime rather than acquire new tokens for each request

Reply:
If we are requesting a token with every request that is significant override head.  Can we run the client through the perf testing framework?

Reply:
Nevermind this should be fine, the tokens are cached in the implementation of DefaultAzureCredential